reactive-messaging-http-quickstart/src/main/java/org/acme/reactivehttp/CostCollector.java

```diff
@@ -0,0 +1,24 @@
+package org.acme.reactivehttp;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+
+@Path("/cost-collector")
+@ApplicationScoped
+public class CostCollector {
+
+    private double sum = 0;
+
+    @POST
+    public void consumeCost(String valueAsString) {
+        sum += Double.parseDouble(valueAsString);
```

Comment:
Concurrency issue - sum can be accessed concurrently.

Reply:
good point, I'll synchronize the stuff, this class does not illustrate the core of the quickstart and synchronization is more readable than while loop for updating an atomic reference